Sources/CacheAdvance/CacheReader.swift

```diff
@@ -50,8 +50,12 @@ final class CacheReader {
         try reader.seek(to: startOffset)
         while let data = try nextEncodedMessage() {
             encodedMessages.append(data)
-            if let endOffset = endOffset, offsetInFile >= endOffset {
-                break
+            if let endOffset = endOffset {
+                if offsetInFile == endOffset {
+                    break
+                } else if offsetInFile > endOffset {
+                    throw CacheAdvanceError.fileCorrupted
+                }
             }
         }
         if let endOffset = endOffset, offsetInFile != endOffset {
```

Comment:
do we need this `if` anymore given the `if` within the `while` loop?

Reply:
This make no difference compared to the code before this. However, this looks more readable to handle `==` and `>` separately. @dfed 

```
                else if offsetInFile > endOffset {
                    throw CacheAdvanceError.fileCorrupted
                }
```

Reply:
Oooh, I misunderstood your question. The `if` within the `while` loop is need to check `offsetInFile == endOffset `  to stop at the end.

Reply:
> Ahh. This code is catching the `offsetInFile < endOffset` case. Maybe we could make that explicit here? Far from necessary, but it could make the intent more clear if I'm reading this right.

I agree it would be nice to make this more clear.

Reply:
I'll take this in a follow-up PR